internal/backend/grpc/management.go

```diff
@@ -654,6 +656,81 @@ func (s *managementServer) ReadAuditLog(req *management.ReadAuditLogRequest, srv
 	return closeFn()
 }
 
+func (s *managementServer) RebootMachine(ctx context.Context, req *management.RebootMachineRequest) (*management.RebootMachineResponse, error) {
+	if _, err := s.authCheckGRPC(ctx, auth.WithRole(role.Admin)); err != nil {
+		return nil, err
+	}
+
+	ctx = actor.MarkContextAsInternalActor(ctx)
+
+	if _, err := safe.StateGetByID[*infra.Machine](ctx, s.omniState, req.MachineId); err != nil {
+		if state.IsNotFoundError(err) {
+			return nil, status.Error(codes.NotFound, "infra machine not found")
+		}
+
+		return nil, err
+	}
+
+	newRebootID := uuid.NewString()
+
+	machineConfig, err := safe.StateGetByID[*omnires.InfraMachineConfig](ctx, s.omniState, req.MachineId)
+	if err != nil && !state.IsNotFoundError(err) {
+		return nil, err
+	}
+
+	if machineConfig == nil {
+		machineConfig = omnires.NewInfraMachineConfig(resources.DefaultNamespace, req.MachineId)
+
+		machineConfig.TypedSpec().Value.RequestedRebootId = newRebootID
+
+		if err = s.omniState.Create(ctx, machineConfig); err != nil {
+			return nil, err
+		}
+	} else {
+		if _, err = safe.StateUpdateWithConflicts(ctx, s.omniState, machineConfig.Metadata(), func(res *omnires.InfraMachineConfig) error {
+			res.TypedSpec().Value.RequestedRebootId = newRebootID
+
+			return nil
+		}); err != nil {
+			return nil, err
+		}
+	}
+
+	eventCh := make(chan safe.WrappedStateEvent[*infra.MachineStatus])
+
+	if err = safe.StateWatch[*infra.MachineStatus](ctx, s.omniState, infra.NewMachineStatus(req.MachineId).Metadata(), eventCh); err != nil {
+		return nil, err
+	}
+
+	var (
+		machineStatus *infra.MachineStatus
+		event         safe.WrappedStateEvent[*infra.MachineStatus]
+	)
```

Comment:
No reason do declare those outside of `for { ... }` since they are not used outside. Actually `if machineStatus, err = event.Resource(); err != nil {` can be split and there will be no need to declare `machineEvent`.